Tools/WebKitTestRunner/ios/UIScriptControllerIOS.mm

```diff
@@ -1295,11 +1320,25 @@ static UIDeviceOrientation toUIDeviceOrientation(DeviceOrientation orientation)
 
 JSRetainPtr<JSStringRef> UIScriptControllerIOS::selectionCaretBackgroundColor() const
 {
-    NSString *serializedColor = webView()._serializedSelectionCaretBackgroundColorForTesting;
-    if (!serializedColor)
-        return { };
+    auto contentView = platformContentView();
+    UIColor *backgroundColor = [contentView valueForKeyPath:@"textInteractionAssistant.selectionView.caretView.backgroundColor"];
+#if HAVE(UI_TEXT_SELECTION_DISPLAY_INTERACTION)
+    if (!backgroundColor) {
+        if (auto view = textSelectionDisplayInteraction().cursorView; !view.hidden)
+            backgroundColor = view.tintColor;
+    }
+#endif
 
-    return adopt(JSStringCreateWithCFString((__bridge CFStringRef)serializedColor));
+    CGFloat red, green, blue, alpha;
+    if (![backgroundColor getRed:&red green:&green blue:&blue alpha:&alpha])
+        return nil;
+
+    NSString *colorDescription;
+    if (alpha == 1)
+        colorDescription = [NSString stringWithFormat:@"rgb(%.0f, %.0f, %.0f)", floor(red * 255), floor(green * 255), floor(blue * 255)];
+    else
+        colorDescription = [NSString stringWithFormat:@"rgba(%.0f, %.0f, %.0f, %.3f)", floor(red * 255), floor(green * 255), floor(blue * 255), alpha];
```

Comment:
Can we reuse `WebCore/ColorSerialization.h` and `WebCore/ColorCocoa.h` here?

Reply:
I tried that at first, but unfortunately ran into some build failures when attempting to link WebCore from WebKitTestRunner. I think it would be doable by exposing a WebCoreTestSupport wrapper instead though, so I'll try this approach and see if it works.